pyanaconda/modules/payload/payload.py

```diff
@@ -42,43 +43,108 @@ def publish(self):
         DBus.publish_object(PAYLOAD.object_path, PayloadInterface(self))
         DBus.register_service(PAYLOAD.service_name)
 
-    def _publish_handler(self):
-        """Publish handler as soon as we know which one to chose"""
-        self._payload_handler.publish()
-
     @property
     def kickstart_specification(self):
         """Return the kickstart specification."""
         return PayloadKickstartSpecification
 
+    @property
+    def payload_handler(self):
+        """Get payload handler.
+
+        Handlers are handling the installation process.
+
+        There are a few types of handler e.g.: DNF, LiveImage...
+        """
+        if self._payload_handler is None:
+            raise HandlerNotSetError()
+        else:
+            return self._payload_handler
+
+    def set_payload_handler(self, payload_handler):
+        """Set payload handler."""
+        self._payload_handler = payload_handler
+        log.debug("Payload handler %s used.", payload_handler.__class__.__name__)
+
+    def get_active_handler_path(self):
+        """Get path of the active payload handler.
+
+        :rtype: str
+        """
+        try:
+            return self.payload_handler.get_handler_path()
+        except HandlerNotSetError:
+            return ""
+
     def process_kickstart(self, data):
         """Process the kickstart data."""
         log.debug("Processing kickstart data...")
 
-        self._create_correct_handler(data)
-        self._publish_handler()
+        handler_class = self._get_correct_handler_class(data)
```

Comment:
I don't think that we need to do this in another method. The method `process_kickstart` is not going to get too big.

Reply:
Fixed by moving this completely to Factory class.